libs/wire-subsystems/src/Wire/ConversationStore/Postgres.hs

```diff
@@ -0,0 +1,1228 @@
+            remoteMems = findMembers convId mParent remotesWithConvId
+        toConv convId localMems remoteMems (Just convRow)
+  pure $ catMaybes mConvs
+  where
+    selectMetadata :: Hasql.Statement [ConvId] [ConvRowWithId]
+    selectMetadata =
+      dimapPG @[_] @(Vector _)
+        @(Vector (_, _, _, Maybe (Vector _), Maybe (Vector _), _, _, _, _, _, _, _, _, _, _, _, _, _))
+        @[ConvRowWithId]
+        [vectorStatement|SELECT (id :: uuid), (type :: integer), (creator :: uuid?), (access :: integer[]?), (access_roles_v2 :: integer[]?),
+                                (name :: text?), (team :: uuid?), (message_timer :: bigint?), (receipt_mode :: integer?), (protocol :: integer?),
+                                (group_id :: bytea?), (epoch :: bigint?), (epoch_timestamp :: timestamptz?), (cipher_suite :: integer?),
+                                (group_conv_type :: integer?), (channel_add_permission :: integer?), (cells_state :: integer?), (parent_conv :: uuid?)
+                         FROM conversation
+                         WHERE id = ANY($1 :: uuid[])
+                        |]
+    selectAllLocalMembers :: Hasql.Statement [ConvId] [LocalMemberRow]
+    selectAllLocalMembers =
+      dimapPG @[_] @(Vector _)
+        [vectorStatement|SELECT (conv :: uuid), ("user" :: uuid), (service :: uuid?), (provider :: uuid?), (otr_muted_status :: integer?), (otr_muted_ref :: text?),
+                                (otr_archived :: boolean?), (otr_archived_ref :: text?), (hidden :: boolean?), (hidden_ref :: text?), (conversation_role :: text?)
+                         FROM conversation_member
+                         WHERE conv = ANY ($1 :: uuid[])
+                         OR conv IN (SELECT parent_conv FROM conversation WHERE id = ANY ($1 :: uuid[]))
+                        |]
+    selectAllRemoteMembers :: Hasql.Statement [ConvId] [RemoteMemberRow]
+    selectAllRemoteMembers =
+      dimapPG @[_] @(Vector _)
+        [vectorStatement|SELECT (conv :: uuid), (user_remote_domain :: text), (user_remote_id :: uuid), (conversation_role :: text)
+                         FROM local_conversation_remote_member
+                         WHERE conv = ANY ($1 :: uuid[])
+                         OR conv IN (SELECT parent_conv FROM conversation WHERE id = ANY ($1 :: uuid[]))
+                        |]
+
+    findMembers :: (HasField "id_" a b, Eq b) => ConvId -> Maybe ConvId -> [(ConvId, a)] -> [a]
+    findMembers convId parentConvId allMembersWithConvId =
+      let localMemsDirect = map snd $ filter (\(memConvId, _) -> memConvId == convId) allMembersWithConvId
+          localMemsParent = map snd $ filter (\(memConvId, _) -> Just memConvId == parentConvId) allMembersWithConvId
+       in nubBy ((==) `on` (.id_)) $ localMemsDirect <> localMemsParent
+
+getLocalConversationIdsImpl :: (PGConstraints r) => UserId -> Maybe ConvId -> Range 1 1000 Int32 -> Sem r (ResultSet ConvId)
```

Comment:
`ResultSet` seems to be a cassandra thing, but i guess this can be cleaned up later.